src/porepy/fracs/utils.py

```diff
@@ -127,3 +127,48 @@ def snap_fracture_set_2d(pts, edges, snap_tol, termination_tol=1e-2, max_iter=10
         logger.warning("Fracture snapping failed to converge")
         logger.warning("Residual: " + str(diff))
         return pts, False
+
+
+def domain_to_dict(domain):
+    """ Transform input to a valid domain dictionary. If domain is already a dictionary,
+    return the dictionary.
+
+    Parameters:
+        domain (None, dict, ndarray, list): Specify the domain either as None, a dictionary,
+        a list domain = [xmax, ymax] (or domain = [xmax, ymax ,zmax])), or a ndarray.
+    Returns:
+        dict: A dictionary. If input domain is None or a dictionary, return intput. If input
```

Comment:
I tried to write down the types here: 

domain_to_dict(domain: typing.Union[typing.List, typing.Dict[str, float], np.ndarray, None]) -> typing.Union[typing.Dict[str, float], None, np.ndarray]:

This is a mess, but it is not clear that it is caused by this PR; most of the problems can be traced way back into the network specification. On the other hand, adding a list did not help.

Reply:
I agree with all of the points here. I wrote this function for FractureNetwork2d and everything was neat and simple. Then I though, this should be easy to include for the 3d case as well. Wrong! There are so many different possibilities to set the boundaries, and they do not agree between 2d and 3d, which caused this mess of I/O arguments. 

Because of this, I think we do not need another way to set the boundary, and I subtract this pull-request. 

However, I am still fed up with making the domain boundary dictionary. Is there a reason we actually want this as a dictionary and not just as a list [xmin, xmax, ymin, ymax....] ?

Reply:
I agree, this is a mess. There is no particular reason for the dict, except that the dictionary makes variable definitions more explicit. This is a real worry to me, I would never be able to remember whether the order is [xmin, xmax, ymin, ymax] or [xmin, ymin, xmax, ymax]. 

How about namedtuples:
from collections import namedtuple
Domain = namedtuple('domain', ['xmin', 'xmax', 'ymin', 'ymax', 'zmin', 'zmax'], defaults=(None, None))
d = Domain(0, 1, -3, 2)
d[0] = 0
d.ymin = -3

One possible issue is that the defaults parameter to namedtuple is new to python3.7, the construction in 3.6 (which I think we will still support for a while, or?) is a bit more cumbersome, see e.g. https://stackoverflow.com/questions/11351032/namedtuple-and-default-values-for-optional-keyword-arguments

This should take care of list and dictionaries. None is still an option, or perhaps we can integrate that into the namedtuple? 

Numpy arrays must still be an option, and I am afraid they will also be introduced to 2d networks at some point, to account for non-box domains there as well. But, if we can go from 4 to 2 formats, where the second option can be considered advanced usage, that is great progress.

Reply:
BoxDomain may be a better name than Domain.